ui/raidboss/emulator/data/LogEventHandler.ts

```diff
@@ -0,0 +1,119 @@
+import EmulatorCommon from '../EmulatorCommon';
+import EventBus from '../EventBus';
+import { LocaleNetRegex } from '../../../../resources/translations';
+import { Lang } from 'types/global';
+import LineEvent from './network_log_converter/LineEvent';
+import { LineEvent0x01 } from './network_log_converter/LineEvent0x01';
+
+export default class LogEventHandler extends EventBus {
+  static doesLineMatch(line: string, regexes: Record<string, RegExp>): RegExpExecArray | false {
+    for (const i in regexes) {
+      const res = regexes[i]?.exec(line);
+      if (res) {
+        if (LocaleNetRegex.areaSeal[i as Lang] && res.groups)
+          res.groups.language = i;
+        return res;
+      }
+    }
+    return false;
+  }
+
+  static isMatchStart(line: string): RegExpMatchArray | false {
+    let res;
+    res = LogEventHandler.doesLineMatch(line, EmulatorCommon.countdownRegexes);
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.StartIn = (parseInt(res.groups.time ?? '0') * 1000).toString();
+      res.groups.StartType = 'Countdown';
+      return res;
+    }
+    res = LogEventHandler.doesLineMatch(line, EmulatorCommon.sealRegexes);
+    if (res) {
+      res.groups = res.groups ?? {};
```

Comment:
```suggestion
    if (res && res.groups) {
```

Reply:
This is an annoying case where `groups` should always be present, but on the off chance that it isn't (due to changes in `LocaleNetRegex` maybe), we'd want to throw an error instead of continuing blindly because it's going to cause issues with the logic in `Encounter` if there's ever a case where there's a start/end line that doesn't have the property set?

Basically, I'm saying that `if (res && !res.groups)` is an error state that should cause the log import process to either halt entirely or skip to the next valid case, and there's not really a good mechanism right now to do the latter.

Reply:
Not depending on the internals of `LocaleNetRegex` is a good argument, thanks.

My followup suggestion is that these should all be `res.groups ??= {}` instead, then.

Reply:
I keep forgetting that `??=` is an operator in typescript. Adjusted for next commit.

Reply:
As a follow up, does it make sense to note this as a comment somewhere to prevent confusion in the future?

Reply:
Yeah, that makes sense.  Some sort of `// Currently all of these regexes have groups if they match at all, but be robust to that changing in the future.` sort of thing?